src/serviceview.cpp

```diff
@@ -33,15 +38,51 @@ void ServiceView::displayTable(QString svcCmdOutput)
     m_dialog->setWindowIcon(m_icon);
     m_dialog->setFixedWidth(600);
     m_dialog->setModal(true);
-    QFormLayout form(m_dialog);
-    QLabel *tableLbl = new QLabel();
-    tableLbl->setOpenExternalLinks(true);
-    tableLbl->setWordWrap(true);
-    tableLbl->setText(svcCmdOutput);
-    form.addRow(tableLbl);
-    QDialogButtonBox buttonBox(Qt::Horizontal, m_dialog);
+    QBoxLayout layout(QBoxLayout::TopToBottom, m_dialog);
+    QJsonParseError error;
+    QJsonDocument json = QJsonDocument::fromJson(svcCmdOutput.toUtf8(), &error);
+    if (json.isNull() || !json.isArray()) {
+        // cannot deseralize the output, print it to the dialog
+        qDebug() << error.errorString();
+        QLabel *tableLbl = new QLabel();
+        tableLbl->setOpenExternalLinks(true);
+        tableLbl->setWordWrap(true);
+        tableLbl->setText(svcCmdOutput);
+        layout.addWidget(tableLbl);
+    } else {
+        // can deseralize the output, generate a table
+        QJsonArray serviceArray = json.array();
+        QTableWidget *table = new QTableWidget(serviceArray.size(), 4);
+        table->resizeColumnsToContents();
+        QStringList headers({ "Namespace", "Name", "Target Port", "URL" });
+        table->horizontalHeader()->setSectionResizeMode(QHeaderView::ResizeToContents);
+        table->horizontalHeader()->setStretchLastSection(true);
+        table->setHorizontalHeaderLabels(headers);
+        for (int i = 0; i < serviceArray.size(); i++) {
+            QJsonObject obj = serviceArray[i].toObject();
+            if (obj.contains("Namespace")) {
+                QTableWidgetItem *namespaceItem = new QTableWidgetItem(obj["Namespace"].toString());
+                table->setItem(i, 0, namespaceItem);
+            }
+            if (obj.contains("Name")) {
+                QTableWidgetItem *nameItem = new QTableWidgetItem(obj["Name"].toString());
+                table->setItem(i, 1, nameItem);
+            }
+            if (obj.contains("PortNames")) {
+                QTableWidgetItem *portNamesItem = new QTableWidgetItem(obj["PortNames"].toString());
```

Comment:
You're doing `toString()` but the `PortNames` field is an array so it's not outputting, same with URLs below

CLI
```
$ minikube service list
|-------------|----------------|--------------|-----|
|  NAMESPACE  |      NAME      | TARGET PORT  | URL |
|-------------|----------------|--------------|-----|
| default     | hello-minikube |         8080 |     |
| default     | kubernetes     | No node port |     |
| kube-system | kube-dns       | No node port |     |
|-------------|----------------|--------------|-----|
```

GUI
<img width="601" alt="Screenshot 2023-07-05 at 2 31 37 PM" src="https://github.com/kubernetes-sigs/minikube-gui/assets/44844360/b14112e1-655b-420c-a930-f7c590919a72">

Reply:
fixed. picture in PR's description is also updated